src/Sarif.Viewer.VisualStudio/Models/CallTree.cs

```diff
@@ -37,34 +42,184 @@ public ObservableCollection<CallTreeNode> TopLevelNodes
             }
         }
 
-        public CodeLocationObject SelectedItem
+
+        public CallTreeNode SelectedItem
         {
             get
             {
-                return _selectedItem;
+                return this._selectedItem;
             }
             set
             {
+                this._selectedItem = value;
+                this.NotifyPropertyChanged(nameof(SelectedItem));
+                
                 // Remove the existing highlighting.
                 if (_selectedItem != null)
                 {
                     _selectedItem.ApplyDefaultSourceFileHighlighting();
                 }
+            }
+        }
 
-                _selectedItem = value;
-                this.NotifyPropertyChanged(nameof(SelectedItem));
+        private static int GetIndexInCallTreeNodeList(IList<CallTreeNode> list, CallTreeNode givenNode)
+        {
+            int index = 0;
+            foreach (CallTreeNode listNode in list)
```

Comment:
>list [](start = 46, length = 4)

Just FYI, you'll raise a NullReferenceException if list is null. Raising the exception seems fine since it's an unexpected condition. But maybe we should validate the argument and throw an ArgumentNullException instead.